src/plugins/graph/models/graph-model.ts

```diff
@@ -119,6 +117,11 @@ export const GraphModel = TileContentModel
       return stringify(snapshot, {maxLength: 200});
     }
   }))
+  .views(self => ({
+    axisShouldShowGridLines(place: AxisPlace) {
+      return self.plotType === 'scatterPlot' && ['left', 'bottom'].includes(place) || !self.data;
```

Comment:
The fact that gridlines should be drawn when there is no data could be controlled by another property available via `useSettingFromStores()` like `emptyPlotIsNumeric`.